tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/PutRecordIntegration.cs

```diff
@@ -0,0 +1,59 @@
+// <copyright file="PutRecordIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Kinesis
+{
+    /// <summary>
+    /// AWSSDK.Kinesis PutRecord CallTarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.Kinesis",
+        TypeName = "Amazon.Kinesis.AmazonKinesisClient",
+        MethodName = "PutRecord",
+        ReturnTypeName = "Amazon.Kinesis.Model.PutRecordResponse",
+        ParameterTypeNames = new[] { "Amazon.Kinesis.Model.PutRecordRequest" },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsKinesisCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class PutRecordIntegration
+    {
+        private const string Operation = "PutRecord";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TPutRecordRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the Kinesis operation</param>
+        /// <returns>CallTarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TPutRecordRequest>(TTarget instance, TPutRecordRequest request)
```

Comment:
I don't see an `OnMethodEnd` in this integration. Is there another method that will be closing the scope that was opened here?

Reply:
Since this is synchronous this may be a .NET Framework-only API, and so you may not have been able to test that. I've got the branch pulled up locally so I can generate the .NET Framework snapshot and make sure that looks okay

Reply:
Since I was able to run and test the .NET Framework code, I added it the `OnMethodEnd` in a87def01baf7fd2c7dfd02c43213b9203138ce0d